homeassistant/components/tesla/__init__.py

```diff
@@ -221,6 +222,12 @@ def unique_id(self) -> str:
     @property
     def icon(self):
         """Return the icon of the sensor."""
+        if self.tesla_device.has_battery():
+            return icon_for_battery_level(
+                battery_level=self.tesla_device.battery_level(),
+                charging=self.tesla_device.battery_charging(),
+            )
+
         return self._icon
```

Comment:
`self._icon` is initialized with an icon that is appropriate for the sensor type. As such, it is used here as a default when the device does not have a battery.